pkg/trait/debug.go

```diff
@@ -35,6 +35,14 @@ func (r *debugTrait) appliesTo(e *Environment) bool {
 	return e.Integration != nil && e.Integration.Status.Phase == v1alpha1.IntegrationPhaseDeploying
 }
 
+func (r *debugTrait) autoconfigure(e *Environment) error {
+	if r.Enabled == nil {
+		enabled := false
+		r.Enabled = &enabled
+	}
+	return nil
+}
+
```

Comment:
Well, the reason why there's a autoconfigure method is mostly because it will be nice in the future to save back in the Integration status (or in the logs) the actual trait configuration used in order to speed-up troubleshooting. Changing it to false in the `autoconfigure` method already allows to determine if the trait code (`apply`) has been executed.

Reply:
yes and no, there's also appliesTo that coul determine if a trait should be applied or not :)

Reply:
We're still Java programmers...

Reply:
we definitively need to review traits ...